src/hotspot/share/opto/node.cpp

```diff
@@ -2047,6 +2049,12 @@ void PrintBFS::print() {
       const Node* n = _print_list.at(i);
       print_node(n);
     }
+    if (_print_igv) {
+      Compile* C = Compile::current();
+      if (C->should_print_igv(0)) {
```

Comment:
I guess the reason to call `should_print_igv` here is to initialize `Compile::_igv_printer` if necessary. Would it be possible to factor out the initialization part https://github.com/openjdk/jdk/blob/2145ace384137b1c028a68dc34a8800577c7a43e/src/hotspot/share/opto/compile.cpp#L5214-L5217 into a separate function that only checks if `_igv_printer` is `nullptr` and, if so, initializes it, and only call that function from here?